### ISSUE TITLE
2.12.0: Tests fail if building with CMake

### ISSUE THREAD

**Issue by Lastique** [gperftools/gperftools#1422]
Here's the build log:

[cmake_fail.txt](https://github.com/gperftools/gperftools/files/12454164/cmake_fail.txt)

The last test is hanging, after a few minutes I interrupted it.

This is on Kubuntu 22.04. On the same machine, gperftools builds and tests pass if using automake.

**Comment by alk** [comment]
Yes, our cmake support is incomplete and best-effort (which in practice is no effort almost)
